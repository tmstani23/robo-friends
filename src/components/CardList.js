import Card from './Card'
import React from 'react';


const CardList = ({robots}) => {
    if(true) {
        throw new Error('ERRORRRR!!');
    }

    return !robots.length > 0 
        ? <h1 className = 'tc'> Loading...</h1>
        : (
            <div>
                {   
                    robots.map((robot, i) => {
                        console.log(robot)
                        return (
                            <Card 
                                key = {i} 
                                id={robot.id} 
                                name={robot.name} 
                                email={robot.email} 
                            />
                        )
            
                    })
                   
                } 
                    
            </div>
            
        )
    
}

export default CardList;