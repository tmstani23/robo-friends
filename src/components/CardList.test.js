import {shallow} from 'enzyme';
import React from 'react';
import CardList from './CardList';
import renderer from 'react-test-renderer';

it('expect card component to render', () => {
    const mockRobots = [
        {
            id: 1,
            name: 'John Snow',
            username: 'John',
            email: 'john@gmail.com'
        }
    ]
    const tree = renderer.create(
        <CardList robots={mockRobots} />
      ).toJSON();
      expect(tree).toMatchSnapshot();
})

it('test alternative return condition', () => {
    const mockRobots = []
    
    const tree = renderer.create(
        <CardList robots={mockRobots} />
      ).toJSON();
    
    expect(tree.children[0].children[0]).toBe(" Loading...")
    
    
});