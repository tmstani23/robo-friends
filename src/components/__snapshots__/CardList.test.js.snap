// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect card component to render 1`] = `
<div>
  <div>
    <div
      className="bg-animate hover-bg-light-blue tc bw2 shadow5 bg-light-green dib br3 pa3 ma2 grow"
    >
      <img
        alt="robots"
        src="https://robohash.org/1?size=200x200"
      />
      <h2>
        John Snow
      </h2>
      <p>
        john@gmail.com
      </p>
    </div>
  </div>
</div>
`;
